services/workers/src/main/java/org/sagebionetworks/trash/worker/TrashWorker.java

```diff
@@ -23,17 +25,19 @@
 	public void run(ProgressCallback progressCallback) {
 		long startTime = System.currentTimeMillis();
 		
-		try{
+		try {
 			List<Long> trashList = trashManager.getTrashLeavesBefore(CUTOFF_TRASH_AGE_IN_DAYS, TRASH_DELETE_LIMIT);
-			logger.info("Purging " + trashList.size() + " entities, older than " +
```

Comment:
Keep going until done, each entity in its own transaction, keep going if one fails.